manifests/server/schema.pp

```diff
@@ -29,20 +30,22 @@
   String[1]            $group            = $postgresql::server::group,
   Stdlib::Absolutepath $psql_path        = $postgresql::server::psql_path,
   Stdlib::Absolutepath $module_workdir   = $postgresql::server::module_workdir,
+  String[1]            $instance         = 'main',
 ) {
   Postgresql::Server::Db <| dbname == $db |> -> Postgresql::Server::Schema[$name]
 
   # If the connection settings do not contain a port, then use the local server port
   $port_override = pick($connect_settings['PGPORT'], $port)
 
   Postgresql_psql {
-    db         => $db,
-    psql_user  => $user,
-    psql_group => $group,
-    psql_path  => $psql_path,
-    port       => $port_override,
-    cwd        => $module_workdir,
+    db               => $db,
+    psql_user        => $user,
+    psql_group       => $group,
+    psql_path        => $psql_path,
+    port             => $port_override,
+    cwd              => $module_workdir,
     connect_settings => $connect_settings,
+    instance         => $instance,
   }
 
   postgresql_psql { "${db}: CREATE SCHEMA \"${schema}\"":
```

Comment:
All the lines similar to 
`postgresql_psql { "${db}: CREATE SCHEMA \"${schema}\"":`
need to have the instance name included, otherwise you won't be able to create the same {schema, role, whatever} in different instances.
Typical error is

```Duplicate declaration: Postgresql_psql[CREATE ROLE foo ENCRYPTED PASSWORD ****] is already declared at....```

Reply:
I am aware of this. We thought about addressing this with an issue for a feature, because having the same objects in different instances on the same node looked like an edge case for us. Would you be OK with this solution?